### PR TITLE
LPS-192879 Emit nonce from server side

### PR DESCRIPTION
**Pull request by izaera**
Sent as draft so that @dsanz can preview it.

### CODE REVIEW

File: modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/taglib/ContentSecurityPolicyTopHeadDynamicInclude.java [hunk 1]
```diff
@@ -48,7 +53,16 @@ public void include(
 		printWriter.print(
 			" type=\"text/javascript\">window.Liferay = window.Liferay || ");
 		printWriter.print("{}; window.Liferay.CSP = {nonce: '");
-		printWriter.print(nonce);
+
```
Comment (dsanz) [inline modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/taglib/ContentSecurityPolicyTopHeadDynamicInclude.java:56]: do we want to emit an empty, but non undefined value for Liferay.CSP object in this case?

File: modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/taglib/ContentSecurityPolicyTopHeadDynamicInclude.java [hunk 2]
```diff
@@ -57,8 +71,35 @@ public void register(DynamicIncludeRegistry dynamicIncludeRegistry) {
 		dynamicIncludeRegistry.register("/html/common/themes/top_head.jsp#pre");
 	}
 
+	private ContentSecurityPolicyConfiguration
+		_getContentSecurityPolicyConfiguration(
+			HttpServletRequest httpServletRequest) {
+
```
Comment (dsanz) [inline modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/taglib/ContentSecurityPolicyTopHeadDynamicInclude.java:77]: This is the same method we have [here](https://github.com/liferay-frontend/liferay-portal/pull/3793/files#diff-fcc978b2491286d2c77bcb694142b53d03d75869d9989f02c2e6d6e505ccfc3aR93), we'll need some sort of utility to host this one

File: modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/taglib/ContentSecurityPolicyTopHeadDynamicInclude.java (hunk 1 above)
Comment (izaera) [reply modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/taglib/ContentSecurityPolicyTopHeadDynamicInclude.java:56]: Yes. An empty nonce means no nonce. We are not using `null` (neither in the client, nor in the server).

**Comment by izaera** [comment]
Closing this to be able to send proper PRs (will send 2 or more)

**Pull request closed without merging**